appinventor/components/src/com/google/appinventor/components/runtime/ChartDataModel.java

```diff
@@ -0,0 +1,554 @@
+     * columns in order of the columns.
+     *
+     * The first element is skipped, since it is assumed that it
+     * is the column name.
+     *
+     * @param columns  columns to import data from
+     */
+    public void importFromColumns(YailList columns) {
+        // Get a YailList of tuples from the specified columns
+        YailList tuples = getTuplesFromColumns(columns);
+
+        // Use the generated tuple list in the importFromList method to
+        // import the data.
+        importFromList(tuples);
+    }
+
+    /**
+     * Constructs and returns a List of tuples from the specified Columns List.
+     * The Columns List is expected to be a List containing Lists, where each
+     * List corresponds to a column, the first entry of which is the header/name
+     * of the column (hence it is skipped in generating data)
+     *
+     * @param columns  List of columns to generate tuples from
+     * @return  Generated List of tuples from the columns
+     */
+    public YailList getTuplesFromColumns(YailList columns) {
+        // Determine the (maximum) row count of the specified columns
+        int rows = ChartDataSourceUtil.determineMaximumListSize(columns);
+
+        List<YailList> tuples = new ArrayList<YailList>();
+
+        // Generate tuples from the columns
+        for (int i = 1; i < rows; ++i) {
+            ArrayList<String> tupleElements = new ArrayList<String>();
+
+            // Add entries to the tuple from all i-th values (i-th row)
+            // of the data columns.
+            for (int j = 0; j < columns.size(); ++j) {
+                Object value = columns.getObject(j);
+
+                // Invalid column specified; Add default value (minus one to
+                // compensate for the skipped value)
+                if (!(value instanceof YailList)) {
+                    tupleElements.add(getDefaultValue(i - 1));
+                    continue;
+                }
+
+                // Safe-cast value to YailList
+                YailList column = (YailList) value;
+
+                if (column.size() > i) { // Entry exists in column
+                    // Add entry from column
+                    tupleElements.add(column.getString(i));
+                } else if (column.size() == 0) { // Column empty (default value should be used)
+                    // Use default value instead (we use an index minus one to componsate
+                    // for the skipped initial value)
+                    tupleElements.add(getDefaultValue(i - 1));
+                } else { // Column too small
+                    // Add blank entry (""), up for the addEntryFromTuple method
+                    // to interpret.
+                    tupleElements.add("");
+                }
+
+                // Alternative solution: Add default values for missing entries as well.
+                // This might, however, not be the desired behavior.
+//                else {
```

Comment:
@ewpatton I feel like that would overcomplicate the importing process overall. Consider the use cases:
* Import from Data File via Source property
* Import from Data File via Source property (Designer)
* Import from Data File via block
* Import from Web via Source property
* Import from Web via Source property (Designer)
* Import from Web via block

For the block parts, it's relatively straight forward -- adding a single extra variable suffices.
However, for the other two use cases, it becomes quite tricky. The Source property is meant to support any Data Source (not just the column-based data sources). For the Designer property I guess it also makes sense to add a variable that gets shown/hidden, but overall the process becomes much more complicated by changing this flag.

Of course an argument could be made for having a flag that is always present and can always be configured, but then this causes a bit of confusion because the property might appear to the users that it does not control anything for that time being, until they realize it can be used for the column-based sources.

So I'm a bit skeptical about this, but I figure we could introduce something of that kind.